File: 1626-can-make-arithmetic-progression-from-sequence/1626-can-make-arithmetic-progression-from-sequence.py
class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        sortedArray = sorted(arr)
        lenArray = len(arr)
        if len == 2:
            return True
        diff = sortedArray[1] - sortedArray[0]
        for i in range(2, lenArray):
            if sortedArray[i] - sortedArray[i-1] != diff:
                return False
        return True
        

File: 1626-can-make-arithmetic-progression-from-sequence/1626-can-make-arithmetic-progression-from-sequence.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        std::sort(arr.begin(), arr.end());

        if (arr.size() == 0) {
            return true;
        }

        int diff = arr[1] - arr[0];
        for (size_t i = 2; i < arr.size(); i++) {
            if(arr[i]-arr[i-1] != diff) {
                return false;
            }
        }
        return true;
        
    }
};